From 865e21cf69eb083cd5413d47d102d0cd75fa94b5 Mon Sep 17 00:00:00 2001
From: Siddharth Manthan <siddharth.manthan@gmail.com>
Date: Thu, 13 May 2021 19:48:21 +0530
Subject: [PATCH 055/210] arm64: dts: qcom: msm8916-samsung-gt58: Add
 touchscreen (v2)

gt58 uses Zinitix bt532 touch controller.
Add it to device tree using Zinitix bt541 drivers.

Signed-off-by: Siddharth Manthan <siddharth.manthan@gmail.com>

v2: Add linux,keycodes
Signed-off-by: Nikita Travkin <nikita@trvn.ru>
---
 .../boot/dts/qcom/msm8916-samsung-gt58.dts    | 55 +++++++++++++++++++
 1 file changed, 55 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt58.dts b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt58.dts
index 2d0935fa160..805c9304fc2 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt58.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt58.dts
@@ -9,6 +9,19 @@ / {
 	compatible = "samsung,gt58", "qcom,msm8916";
 	chassis-type = "tablet";
 
+	reg_vdd_tsp: regulator-vdd-tsp {
+		compatible = "regulator-fixed";
+		regulator-name = "vdd_tsp";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+
+		gpio = <&msmgpio 73 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&reg_tsp_en_default>;
+	};
+
 	vibrator {
 		compatible = "gpio-vibrator";
 		enable-gpios = <&msmgpio 76 GPIO_ACTIVE_HIGH>;
@@ -18,7 +31,49 @@ vibrator {
 	};
 };
 
+&blsp_i2c5 {
+	status = "okay";
+
+	/*
+	 * Actual IC is Zinitix bt532.
+	 * It seems compatible enough with bt541 for basic functionality.
+	 */
+
+	touchscreen@20 {
+		compatible = "zinitix,bt532", "zinitix,bt541";
+		reg = <0x20>;
+		interrupt-parent = <&msmgpio>;
+		interrupts = <13 IRQ_TYPE_EDGE_FALLING>;
+
+		touchscreen-size-x = <768>;
+		touchscreen-size-y = <1024>;
+		linux,keycodes = <KEY_APPSELECT KEY_BACK>;
+
+		vdd-supply = <&reg_vdd_tsp>;
+		vddo-supply = <&pm8916_l6>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&tsp_int_default>;
+	};
+};
+
 &msmgpio {
+	reg_tsp_en_default: reg-tsp-en-default {
+		pins = "gpio73";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
+	tsp_int_default: tsp-int-default {
+		pins = "gpio13";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
 	vibrator_en_default: vibrator-en-default {
 		pins = "gpio76";
 		function = "gpio";
-- 
2.34.1

